addons/checkparam/checkparam.lua

```diff
@@ -0,0 +1,174 @@
+    job_define = settings[windower.ffxi.get_player().main_job:lower()]
+    return job_define and windower.regex.split(job_define,'[|]')
+        or {'store tp','quadruple attack','triple attack','double attack','',
+            'magic accuracy','fast cast','haste','dual wield','',
+            'magic attack bonus','magic burst damage','magic damage','',
+            'enmity','cure potency'}
+end
+function split_text(txt,id,arg)
+    for key,value in string.gmatch(txt,"/?([%D]-):?([%+%-]?[0-9]+)%%?%s?") do
+        local key = windower.regex.replace(string.lower(key),'(\\"|\\.|\\s$)','')
+        local key = integrate[key] or key
+        local key = arg and arg..key or key
+        tbl[key] = tonumber(value) + (tbl[key] or 0)
+        if debug then
+            local color = windower.regex.match(key,'enhanced') and 170
+            windower.add_to_chat(color or 72,'['..id..']'..res.items[id].en..'['..key..'] '..value..' '..tbl[key])
+        end
+    end
+end
+function checkparam(arg1)
+    debug = arg1=='debug'
+    tbl,txt={},
+    windower.send_command('input /checkparam <me>')
+    for slot,bag in pairs(windower.ffxi.get_items().equipment) do
+        gtbl = windower.ffxi.get_items().equipment
+        if not string.find(slot,'_bag') then
+            gear = windower.ffxi.get_items(gtbl[slot..'_bag'],gtbl[slot])
+            if gtbl[slot] > 0 and gear.id < 65535 then
+                if res.item_descriptions[gear.id] then
+                    local txt = windower.regex.split(res.item_descriptions[gear.id].en,"(Pet|Avatar|Automaton|Wyvern|Luopan): ")
+                    for i,v in pairs(windower.regex.split(txt[1],"\n")) do
+                        split_text(v,gear.id)
+                    end
+                    if txt[2] then
+                        local txt = windower.regex.replace(txt[2],'\n',' ')
+                        split_text(txt,gear.id,'pet: ')
+                    end
+                end
+                if gear.extdata and extdata.decode(gear).type == 'Augmented Equipment' then
+                    for slot,augment in pairs(extdata.decode(gear).augments) do
+                        split_text(augment,gear.id)
+                    end
+                end
+                if enhanced[res.items[gear.id].en] then
+                    local txt = string.split(string.gsub(enhanced[res.items[gear.id].en], "([+-:][0-9]+)",",%1"),",")
+                    tbl[txt[1]] = tonumber(txt[2]) + (tbl[txt[1]] or 0)
+                    if debug then
+                        windower.add_to_chat(170,'['..gear.id..']'..res.items[gear.id].en..'['..txt[1]..'] '..txt[2]..' '..tbl[txt[1]])
+                    end
+                end
+            end
+        end
+    end
+    windower.add_to_chat(69, '//checkparam')
+    for index,key in pairs(key_define()) do
+        txt = key == '' and txt..'\n' or (txt or '')..'['..key..'] '..(tbl[string.lower(key)] or 0)..' '
+    end
+    windower.add_to_chat(70, txt)
+    collectgarbage()
```

Comment:
Any particular reason why you manually collect garbage here?

Reply:
because the 'lua memory' after checkparam() was small. is that meaningless?